asp_interactive_configuration/ooasp/encodings/ooasp_aux_kb.lp

```diff
@@ -32,19 +32,23 @@ ooasp_leafclass(C) :-
      ooasp_class(C),
      not ooasp_subclass(_,C).
 
-% Defines the limits for an association based on the class number
 
 % ooasp_assoc_limit(ASSOC,min,1,C1,C2MIN)
-% The association ASSOC must be from a C1 element to at least C2MIN elements
+% The association ASSOC must be from a C1 object to at least C2MIN objects
 
-ooasp_assoc_limit(ASSOC,min,1,C1,C2MIN):-
+ooasp_assoc_limit(ASSOC,min,1,C1,C2MIN,C2):-
 	ooasp_assoc(ASSOC,C1,C1MIN,C1MAX,C2,C2MIN,C2MAX).
 
-ooasp_assoc_limit(ASSOC,max,1,C1,C2MAX):-
+ooasp_assoc_limit(ASSOC,max,1,C1,C2MAX,C2):-
 	ooasp_assoc(ASSOC,C1,C1MIN,C1MAX,C2,C2MIN,C2MAX).
 
-ooasp_assoc_limit(ASSOC,min,2,C2,C1MIN):-
+ooasp_assoc_limit(ASSOC,min,2,C2,C1MIN,C1):-
 	ooasp_assoc(ASSOC,C1,C1MIN,C1MAX,C2,C2MIN,C2MAX).
 
-ooasp_assoc_limit(ASSOC,max,2,C2,C1MAX):-
+ooasp_assoc_limit(ASSOC,max,2,C2,C1MAX,C1):-
 	ooasp_assoc(ASSOC,C1,C1MIN,C1MAX,C2,C2MIN,C2MAX).
+
+
+is_an_assoc_specialization(ASSOC_S):- ooasp_assoc_specialization(ASSOC_S,ASSOC).
+specialization_priority(ASSOC,1):-is_an_assoc_specialization(ASSOC).
+specialization_priority(ASSOC,20):-not is_an_assoc_specialization(ASSOC), ooasp_assoc(ASSOC,_,_,_,_,_,_).
```

Comment:
The predicate `specialization_priority` seems never to be used.

Reply:
I removed them